### PR TITLE
Annotation cleanup

### PR DESCRIPTION
**Pull request by Dananji**
Related issue: #759

### CODE REVIEW

**Comment by Dananji** [comment]
This PR includes:
- cleanup of playlist marker parsing and adjustments to annotation parsing code to include these annotations
- cleanup in transcript-parser of redundant code for parsing transcript information in Manifests
- re-purpose annotations parsed and stored in global state by annotations component in transcript component instead of re-parsing the Manifest

I tested this in Avalon for markers display and edit actions to make sure these changes doesn't break the functionality in Avalon for playlists.

We will need to test manifest from different producers (Avalon, Aviary, AVAnnotate) to ensure none of the functionalities in transcript and annotation components are not broken.

**Review by cjcolvar — COMMENTED** [review]
Looks great overall!  All of the comments are super helpful especially noting where assumptions have been made.  I had a few questions about where some code should live but I don't think they are blockers.

File: src/components/MarkersDisplay/Annotations/AnnotationSetSelect.js [hunk 1]
```diff
@@ -25,9 +26,10 @@ const AnnotationSetSelect = ({
 
     if (canvasAnnotationSets?.length > 0) {
```
[the remark this answers is not in the document]
Comment (Dananji) [reply src/components/MarkersDisplay/Annotations/AnnotationSetSelect.js:27]: `AnnotationSetSelect` component does not get rendered by the `AnnotationsDisplay` component when `canvasAnnotationSets` is an empty set. But it would be cleaner to reset `timedAnntoationSets` in this case in `AnnotationSetSelect` component. I will make that change 👍

File: src/components/MarkersDisplay/MarkersDisplay.js [hunk 2]
```diff
@@ -46,29 +47,44 @@ const MarkersDisplay = ({
...
+        let canvasMarkers = [];
+        // Filter all markers from annotationSets for the current Canvas
+        if (annotationSets?.length > 0) {
+          canvasMarkers = annotationSets.map((a) => a.markers)
+            .filter(m => m != undefined).flat();
+        }
+        // Update markers in local and global state
+        manifestDispatch({ markers: { canvasIndex, canvasMarkers }, type: 'setPlaylistMarkers' });
+        setCanvasMarkers(canvasMarkers);
+      }
```
[the remark this answers is not in the document]
Comment (Dananji) [reply src/components/MarkersDisplay/MarkersDisplay.js:79]: Yes, that sounds great! I'll make that change here.

File: src/services/ramp-hooks.js [hunk 3]
```diff
@@ -456,10 +471,8 @@ export const useVideoJSPlayer = ({
       }
 
       let playlistMarkers = [];
-      if (playlist?.markers?.length) {
-        const canvasMarkers = playlist.markers
-          .filter((m) => m.canvasIndex === canvasIndex)[0].canvasMarkers;
-        playlistMarkers = canvasMarkers.map((m) => ({
+      if (markers?.length > 0) {
+        playlistMarkers = markers.map((m) => ({
```
[the remark this answers is not in the document]
Comment (Dananji) [reply src/services/ramp-hooks.js:475]: No, this creates a list of playlist markers in the format the `videojs-markers-plugin` expects from `markers` array built in the above `useMemo`.

**Review by cjcolvar — APPROVED** [review]
Looks great! :rocket:
